File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> rank, vector<bool> attendance) {
    vector<pair<int, int>> answer;
    for(int i = 0; i < rank.size(); i++){
        if(attendance[i]) answer.push_back(make_pair(rank[i], i));
    }
    sort(answer.begin(), answer.end());
    answer.resize(3);
    return 10000 * answer[0].second + 100 * answer[1].second + answer[2].second;
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.js
function solution(rank, attendance) {
    const pos = Array(rank.length);
    for(let i = 0; i < rank.length; i++) pos[rank[i]] = i;

    const picked = [];
    for (let r = 1; r <= rank.length && picked.length < 3; r++) {
      const i = pos[r];
      if (attendance[i]) picked.push(i);
    }
    const [a, b, c] = picked;
    return 10000 * a + 100 * b + c;
    
}